intel-types/src/lib.rs

```diff
@@ -19,24 +22,48 @@ use testing::testaso;
 /// the struct and its mask for simplicity.
 #[repr(C)]
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub struct Masked<T: Copy + Debug + PartialEq + BitAnd<Output = T>> {
+pub struct Masked<T: BitAnd<Output = T>> {
     /// The data being masked, e.g. Attribute flags.
     pub data: T,
 
     /// The mask.
     pub mask: T,
 }
 
-impl<T: Copy + Debug + PartialEq + BitAnd<Output = T>> From<T> for Masked<T> {
+impl<T> Default for Masked<T>
+where
+    T: BitAnd<Output = T>,
+    T: BitOr<Output = T>,
+    T: Not<Output = T>,
+    T: Default,
+    T: Copy,
+{
+    fn default() -> Self {
+        T::default().into()
+    }
+}
+
+impl<T> From<T> for Masked<T>
+where
+    T: BitAnd<Output = T>,
+    T: BitOr<Output = T>,
+    T: Not<Output = T>,
+    T: Copy,
+{
     fn from(value: T) -> Self {
         Self {
             data: value,
-            mask: value,
+            mask: value | !value,
```

Comment:
This is bitwise or? So the mask is by default all 1's, right (nothing is masked out)?

Reply:
Basically, we need a way to say "all flags". However, `bitflags` doesn't implement any trait for this. So instead we calculate it by a bitwise or of all the flags in the value and all the flags not in the value.